crates/h264/src/sps.rs

```diff
@@ -53,24 +648,63 @@ impl Sps {
             return Err(io::Error::new(io::ErrorKind::InvalidData, "Forbidden zero bit is set"));
         }
 
-        bit_reader.seek_bits(2)?; // nal_ref_idc
-
-        let nal_unit_type = bit_reader.read_bits(5)?;
+        let nal_ref_idc = bit_reader.read_bits(2)? as u8;
+        let nal_unit_type = bit_reader.read_bits(5)? as u8;
         if nal_unit_type != 7 {
```

Comment:
We should do the conversion to `NALUnitType` here and then check `NALUnitType::SPS` below